### PR TITLE
Fix false negatives such as "mÃ¡s"

### PR DESCRIPTION
**Pull request by rspeer**
This regression has been around for a while and was reported in #88. Our heuristics were conservative around certain punctuation characters, causing them to not be fixed in the obvious cases such as `mÃ¡s` for "más".

The new heuristic also lets me enable one previously disabled test case.

While testing borderline cases, I came across a potential false positive that has probably existed for as long as ftfy has. I added a disabled test for it to take note.

### CODE REVIEW

**Comment by rspeer** [comment]
Oh right. One other change I'm making here is that, given that I'm making the "badness" regexes a little more complex, I can save some running time by simplifying the ones that exist. There's no reason they needed capturing groups.

File: tests/test_cases.json [hunk 1]
```diff
@@ -420,5 +425,18 @@
...
+        "original": "NICIODATĂ\u00a0",
+        "fixed": "NICIODATĂ\u00a0",
+        "enabled": true
+    },
+    {
+        "label": "Synthetic, false positive: the Sami name of a town in Lapland, in capital letters, followed by a non-breaking space",
+        "comment": "Related testing tells me I should worry about cases like this, though I haven't seen a real example",
+        "original": "SUOVÂKUOŠKÂ\u00a0",
+        "fixed": "SUOVÂKUOŠKÂ\u00a0",
+        "enabled": false
```
Comment (alin-luminoso) [inline tests/test_cases.json]: Not that this would want to be fixed in this PR, but as a drive-by: I wonder if this file would be easier to read if `"enabled"` were considered `true` by default.

I also wonder if better than disabling the case would be, say, `"current": <whatever ftfy does now>`, which would override `"fixed"` but still let the test case run.  This way you would document what it's doing instead and be alerted if that behavior changed -- hopefully to the correct behavior, but presumably you want to find out anyway.

File: ftfy/badness.py [hunk 2]
```diff
@@ -35,9 +35,6 @@ def _make_weirdness_regex():
     """
     groups = []
 
-    # Match lowercase letters that are followed by non-ASCII uppercase letters
-    groups.append('lA')
```
Comment (alin-luminoso) [inline ftfy/badness.py]: Just out of curiosity, why this change?

Comment (rspeer) [reply ftfy/badness.py]: It's not even doing what it says, and is redundant with `[Ll][AaC]` a few lines down.

Comment (rspeer) [reply ftfy/badness.py]: But yes, good that you asked, as I didn't describe this change.

Comment (alin-luminoso) [reply ftfy/badness.py]: Gosh, it sure is.  (Is redundant with below, I mean.)  Cool beans.

File: tests/test_cases.json (hunk 1 above)
Comment (rspeer) [reply tests/test_cases.json]: I wanted the test case data to be reasonably explicit, but I can see why this would help -- it would make the disabled test cases useful for more than bookkeeping, and it'd be particularly desirable so we can notice if a case we're not fixing happens to become fixed.
